services/storage/src/simcore_service_storage/dsm.py

```diff
@@ -491,15 +491,57 @@ async def try_update_database_from_storage(
     )
     async def auto_update_database_from_storage_task(
         self, file_uuid: str, bucket_name: str, object_name: str
-    ):
+    ) -> Optional[FileMetaDataEx]:
         return await self.try_update_database_from_storage(
             file_uuid, bucket_name, object_name, silence_exception=True
         )
 
-    async def upload_link(self, user_id: str, file_uuid: str, as_presigned_link: bool):
+    async def update_metadata(
```

Comment:
so update here refers to "re-sync" all metadata in pg with s3 storage, right? VERY NICE 🎉

Reply:
I've dropped this delete_metadata since it made no sense to support it. Actually calling the endpoint to remove the file works as expected.